tensorboard/plugins/hparams/tf_hparams_query_pane/tf-hparams-query-pane.html

```diff
@@ -447,6 +447,24 @@
           notify: true,
         },
 
+        // We track both "data found" and "data not found", because this
+        // makes it expedient to display nothing until the initial fetch is
+        // complete.  In the absence of this "undefined" case, we would end up
+        // briefly displaying the "no data" message before the data arrives.
+        // Note that, in the context of a dom-if condition, we cannot easily
+        // distinguish between false and undefined.
+        dataFound: {
+          type: Boolean,
+          value: false,
+          notify: true,
+        },
+
+        noDataFound: {
+          type: Boolean,
+          value: false,
+          notify: true,
+        },
```

Comment:
Hm, could we avoid confusing cases (`noDataFound === false`) if we had:

- hasHparamsOrMetrics: default false, becomes true if the latest response has 1+ hparams/metrics
- hparamsOrMetricsDataLoaded: default false, becomes true after receiving our first response

?

Reply:
That would indeed be clearer here.  The problem arises where these are used, in `tf-hparams-main`, where we have things like `<template is="dom-if" if="[[_noDataFound]]">`.  The issue is that the condition of `dom-if` can be a concrete property or a function call, or the negation of those, but not any more complicated expression.  Thus, if we use the clearer state properties here, we'd have to introduce functions like `dataFound(hasHparamsOrMetrics, hparamsOrMetricsDataLoaded) { return hasHparamsOrMetrics && hparamsOrMetricsDataLoaded; }`.  Preparing the properties concretely here is a bit hacky, but avoids that additional complication.

Added a comment about the confusing case; hope that helps.

Reply:
Ah, Polymer template expressions are so limited :/

The extra complexity of `dataFound(hasHparamsOrMetrics, hparamsOrMetricsDataLoaded)` seems reasonable to me.  However if you prefer the simpler 2 properties, could we rename them to something more explicit?
e.g.
`dataFound` --> `hasNonEmptyHparamsOnDataLoaded` or `dataLoadedWithNonEmptyHparams`
`noDataFound` --> `hasEmptyHparamsOnDataLoaded` or `dataLoadedWithEmptyHparams`

or similar would be much less confusing to me, esp in tf-hparams-main.html.

Reply:
Sure, the renaming sgtm; done.